chemprop/v2/data/datasets.py

```diff
@@ -305,3 +305,41 @@ def smiles(self) -> list[str]:
     @property
     def mols(self) -> list[Chem.Mol]:
         return [(d.rct, d.pdt) for d in self.data]
+
+
+@dataclass(repr=False, eq=False)
```

Comment:
`repr` because the default representation of a multicomponent dataset is horribly ugly and doesn't provide any parseable information to a user. `eq` because we're not really comparing different `MulticomponentDataset`s. Even if we were the default implementation would be insufficient.

As to why I've just set these for `MulticomponentDataset`, I could also set them for `{Molecule,Reaction}Dataset` too reasonably, but I just didn't get around to it.